src/components/common/BottomSheet.jsx

```diff
@@ -0,0 +1,29 @@
+import React, { useRef } from "react";
+import { IoMdClose } from "react-icons/io";
+import useOnClickOutside from "../../hooks/useOnClickOutside";
+
+export default function BottomSheet({ handler, children }) {
+  const modalRef = useRef(null);
+  const modalWrapperRef = useRef(null);
+
+  useOnClickOutside(modalRef, modalWrapperRef, handler);
+
+  return (
+    <>
+      <div
+        ref={modalWrapperRef}
+        className="modal_wrapper w-full h-full bg-black flex items-center justify-center absolute opacity-30 z-10"
+      />
```

Comment:
바텀시트의 바깥쪽 반투명한 곳을 눌렀을 때 이벤트를 동작시키기 위해 ref랑 useOnClickOutside 훅을 만들어주신 것 같아요
그런데 코드 상으로는 modalWrapperRef 안쪽에 modalRef가 있는 형태가 아니여서 modalRef에 클릭 이벤트가 발생해도 modalWrapperRef에 전달되지 않을 거예요.
그래서 ref와 훅을 사용하지 않고, modalWrapperRef를 부여한 div 태그에 onClick 이벤트를 달아주는게 좋을 것 같습니다!
해보고 안 되면 질문해주세요!

```suggestion
  return (
    <>
      <div
        onClick={handler}
        className="modal_wrapper w-full h-full bg-black flex items-center justify-center absolute opacity-30 z-10"
      />
```

Reply:
커스텀 훅에서 동작하도록 만들었습니다!